### PR TITLE
#29 rendered todos list to html page

### PR DESCRIPTION
**Pull request by TurnbulA**
Renders the title line of the json to html

### CODE REVIEW

File: todos.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+const renderJson = json => {
+  el.innerHTML = json.title;
+  const userTasks = json
+    .map(
+      x => `<ul class="list">
+  <li class ="json-section__userid">${x.userId}</li>
+  <li class ="json-section__id">${x.id}</li>
+  <li class ="json-section__title">${x.title}</li>
+  <li class ="json-section__completed">${x.completed}</li>
+  <br>
```
Comment (mali891) [inline todos.js]: Dont do this, use CSS margins

File: todos.js [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+fetch("https://jsonplaceholder.typicode.com/todos")
   .then(response => response.json())
-  .then(json => (el.innerHTML = json.title));
+  .then(json => renderJson(json));
+
+const renderJson = json => {
+  el.innerHTML = json.title;
+  const userTasks = json
+    .map(
+      x => `<ul class="list">
```
Comment (mali891) [inline todos.js]: `userTask`

File: todos.js [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+  <li class ="json-section__userid">${x.userId}</li>
+  <li class ="json-section__id">${x.id}</li>
+  <li class ="json-section__title">${x.title}</li>
+  <li class ="json-section__completed">${x.completed}</li>
+  <br>
+  </ul>`
+    )
+    .join("");
+  el.innerHTML = userTasks;
```
Comment (mali891) [inline todos.js:22]: Leave a line break here

File: todos.js (hunk 2 above)
Comment (mali891) [inline todos.js]: Put the `ul` onto a new line and format the template string as it would look in the html

File: todos.js [hunk 4]
```diff
@@ -2,7 +2,9 @@ const el = document.querySelector(".json-section");
 
 fetch("https://jsonplaceholder.typicode.com/todos")
   .then(response => response.json())
-  .then(json => renderJson(json));
+  .then(json => renderJson(json))
+  .catch(error => error(error))
+  .finally(() => alert("Done"));
```
Comment (mali891) [inline todos.js]: Unnecessary

File: todos.js [hunk 5]
```diff
@@ -2,7 +2,9 @@ const el = document.querySelector(".json-section");
 
 fetch("https://jsonplaceholder.typicode.com/todos")
   .then(response => response.json())
-  .then(json => renderJson(json));
+  .then(json => renderJson(json))
+  .catch(error => error(error))
```
Comment (mali891) [inline todos.js]: `error` is not a function

File: todos.js (hunk 4 above)
Comment (stephen-algeo) [reply todos.js]: more alerts

File: todos.js (hunk 5 above)
Comment (stephen-algeo) [reply todos.js]: lookup `console.error()`

File: todos.js [hunk 6]
```diff
@@ -3,8 +3,7 @@ const el = document.querySelector(".json-section");
 fetch("https://jsonplaceholder.typicode.com/todos")
   .then(response => response.json())
   .then(json => renderJson(json))
-  .catch(error => error(error))
-  .finally(() => alert("Done"));
+  .catch(console.error(error));
```
Comment (mali891) [inline todos.js]: `.catch(error => console.error(error))`

**Review by mali891 — APPROVED** [review]

**Pull request merged**